src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -117,22 +138,29 @@ public bool IsEmpty
         /// <param name="work">The WorkItem to process</param>
         public void Enqueue(WorkItem work)
         {
-            lock (_syncRoot)
+            do
```

Comment:
There is no need for a backing delay, since CAS automatically waits for the "simultaneous" operations to complete (the atomicity of the operation).